packages/block-editor/src/components/off-canvas-editor/index.js

```diff
@@ -183,34 +187,59 @@ function __ExperimentalOffCanvasEditor(
 
 	return (
 		<AsyncModeProvider value={ true }>
-			<ListViewDropIndicator
-				listViewRef={ elementRef }
-				blockDropTarget={ blockDropTarget }
-			/>
-			<TreeGrid
-				id={ id }
-				className="block-editor-list-view-tree"
-				aria-label={ __( 'Block navigation structure' ) }
-				ref={ treeGridRef }
-				onCollapseRow={ collapseRow }
-				onExpandRow={ expandRow }
-				onFocusRow={ focusRow }
-				applicationAriaLabel={ __( 'Block navigation structure' ) }
-			>
-				<ListViewContext.Provider value={ contextValue }>
-					<ListViewBranch
-						blocks={ clientIdsTree }
-						selectBlock={ selectEditorBlock }
-						showBlockMovers={ showBlockMovers }
-						fixedListWindow={ fixedListWindow }
-						selectedClientIds={ selectedClientIds }
-						isExpanded={ isExpanded }
-						shouldShowInnerBlocks={ shouldShowInnerBlocks }
-						selectBlockInCanvas={ selectBlockInCanvas }
-					/>
-				</ListViewContext.Provider>
-			</TreeGrid>
+			<div className="offcanvas-editor">
+				<ListViewDropIndicator
+					listViewRef={ elementRef }
+					blockDropTarget={ blockDropTarget }
+				/>
+				<TreeGrid
+					id={ id }
+					className="block-editor-list-view-tree"
+					aria-label={ __( 'Block navigation structure' ) }
+					ref={ treeGridRef }
+					onCollapseRow={ collapseRow }
+					onExpandRow={ expandRow }
+					onFocusRow={ focusRow }
+					applicationAriaLabel={ __( 'Block navigation structure' ) }
+				>
+					<ListViewContext.Provider value={ contextValue }>
+						<ListViewBranch
+							blocks={ clientIdsTree }
+							selectBlock={ selectEditorBlock }
+							showBlockMovers={ showBlockMovers }
+							fixedListWindow={ fixedListWindow }
+							selectedClientIds={ selectedClientIds }
+							isExpanded={ isExpanded }
+							shouldShowInnerBlocks={ shouldShowInnerBlocks }
+							selectBlockInCanvas={ selectBlockInCanvas }
+						/>
+					</ListViewContext.Provider>
+					<tr>
+						<BlockListAppender
+							tagName="td"
+							rootClientId={ clientId }
+							renderAppender={ () => (
+								<OffCanvasEditorAppender
+									rootClientId={ clientId }
+								/>
+							) }
+						/>
+					</tr>
+				</TreeGrid>
+			</div>
 		</AsyncModeProvider>
 	);
 }
+
+function OffCanvasEditorAppender( { rootClientId } ) {
+	return (
+		<ButtonBlockAppender
+			className="offcanvas-editor__list-appender__toggle"
+			rootClientId={ rootClientId }
+		>
+			{ __( 'Add Menu Item' ) }
```

Comment:
Annoyingly the internal implementation of `ButtonBlockAppender` doesn't allow us to customise the label. So we have one text based label here and another provided by `ButtonBlockAppender`.